jme3-core/src/main/java/com/jme3/anim/tween/action/Action.java

```diff
@@ -33,38 +127,62 @@ public double getLength() {
     /**
      * Alter the length (duration) of this Action.  This can be used to extend
      * or truncate an Action.
-     *
      * @param length the desired length (in unscaled seconds, default=0)
      */
     public void setLength(double length) {
         this.length = length;
     }
 
+    /**
+     * Gets the speed of the frames of this action which gets applied for the Layer enclosing this action.
+     * Gets applied each advance() by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * @return the speed of frames.
+     */
     public double getSpeed() {
         return speed;
     }
 
+    /**
+     * Sets the speed of the frames of this action which gets applied for the Layer enclosing this action.
+     * Gets applied each advance() by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * @param speed the speed of frames.
+     */
     public void setSpeed(double speed) {
         this.speed = speed;
-        if( speed < 0){
-            setForward(false);
-        } else {
-            setForward(true);
-        }
+        setForward(speed < 0);
     }
 
+    /**
+     * Gets the animation mask for this action.
+     * the animation mask holds data(subsets of an object) about which part of the animation would be played.
+     * @return the animation mask instance.
+     */
     public AnimationMask getMask() {
         return mask;
     }
 
+    /**
+     * Sets the animation mask for this action.
+     * the animation mask holds the info about which part(subsets of an object) of the animation to play, like joints in #{@link com.jme3.anim.ArmatureMask#addFromJoint(Armature, String)}
+     * @param mask the animation mask instance.
+     */
     public void setMask(AnimationMask mask) {
         this.mask = mask;
     }
 
+    /**
+     * Checks for the forward boolean flag.
```

Comment:
"Checks for" implies that the flag might not exist. Change "Checks for" to "Tests" or "Tests the value of".
Also, "boolean" is redundant with "flag". Use one word or the other, not both.

Reply:
I will change `Checks for` to `tests the value of`.

In my code (and on other APIs like android/pi4j), sometimes `enum` values are used as flags (ie : to trigger something to behave based on a value), that's why i have used this terminology.